hikari/__init__.py

```diff
@@ -72,11 +74,11 @@
 from hikari.files import Rawish
 from hikari.files import Resourceish
 from hikari.guilds import *
-from hikari.impl.bot import BotApp
-from hikari.impl.bot import GatewayBot
-from hikari.impl.rest import ClientCredentialsStrategy
-from hikari.impl.rest import RESTApp
-from hikari.impl.rest_bot import RESTBot
+from hikari.impl import BotApp
+from hikari.impl import ClientCredentialsStrategy
+from hikari.impl import GatewayBot
+from hikari.impl import RESTApp
+from hikari.impl import RESTBot
```

Comment:
This import method is not something we have used until now. We always have the full qualified path to to object (how it was before)

Reply:
Except you kinda don't in `__init__.py` files, in those you do stuff like `hikari.events import *` to import from hikari.events and stuff from in modules within hikari.events which is just an equivalent

Reply:
Also what you described there is kinda unnecessary and if anything possibly even slow
Doing the following

```py
from hikari.events import channel_events
from hikari.events import guild_events
from hikari.events import interaction_events
from hikari.events import lifetime_events
from hikari.events import member_events
from hikari.events import message_events
from hikari.events import reaction_events
from hikari.events import role_events
from hikari.events import shard_events
from hikari.events import typing_events
from hikari.events import user_events
from hikari.events import voice_events
```

when you can just do while accesing the event types on events instead of their specific modules

```py
from hikari.events
```

honestly seems like a lot of unnecessary import calls